lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1142,6 +1142,10 @@ def _post_start_init(self, event: _StartOpenSearch):  # noqa: C901
         if self.opensearch_peer_cm.is_provider():
             self.peer_cluster_provider.refresh_relation_data(event, can_defer=False)
 
+        # before resetting the CA rotation state, we remove the old ca from the truststore
+        if self.peers_data.get(Scope.UNIT, "tls_ca_renewed", False):
+            self.tls.remove_old_ca()
```

Comment:
This should only happen if all nodes / all applications have finished rotating the certificates after a CA rotation. I think we need an additional flag on the peer-cluster relation data if it's a large deployment, no?

Reply:
Yes, currently the `old-ca` would be removed from the truststore even before all cluster-members have updated the new certs. But the truststore file would only be loaded on another restart, so it does not get active immediately.